dist/src/main/java/io/camunda/zeebe/shared/management/ClusterEndpoint.java

```diff
@@ -162,6 +169,116 @@ private ResponseEntity<?> scaleBrokers(
     }
   }
 
+  @PatchMapping(consumes = "application/json", produces = "application/json")
+  public ResponseEntity<?> updateClusterConfiguration(
+      @RequestParam(defaultValue = "false") final boolean dryRun,
+      @RequestParam(defaultValue = "false") final boolean force,
+      @RequestBody final ClusterConfigPatchRequest request) {
+    try {
+      final var brokers = request.getBrokers();
+      final var partitions = request.getPartitions();
+      if (force) {
+        return forceRemoveBrokers(dryRun, brokers, partitions);
+      }
+
+      if (brokers != null
+          && brokers.getCount() != null
+          && ((brokers.getAdd() != null || !brokers.getAdd().isEmpty())
+              || (brokers.getRemove() != null || !brokers.getRemove().isEmpty()))) {
+        return invalidRequest(
+            "Cannot change brokers count and add/remove brokers at the same time. Specify either the newPartitionCount or brokers to add and remove.");
+      }
+
+      final Optional<Integer> newPartitionCount =
+          partitions != null ? Optional.ofNullable(partitions.getCount()) : Optional.empty();
+      final Optional<Integer> newReplicationFactor =
+          partitions != null
+              ? Optional.ofNullable(partitions.getReplicationFactor())
+              : Optional.empty();
```

Comment:
:upside_down_face: can also be expressed as `Optional.ofNullabel(Partitions).map(ClusterConfigPatchRequestPartitions::getReplicationFactor)`